### PR TITLE
PROD-72: Login / Sign up + bootstrap

### PR DESCRIPTION
**Pull request by Pavliv**
# Reviewers

- [x] @vikabelous
- [x] @okohuch
- [x] @gnaizer 
- [x] @sashamuliar

# Jira task
https://ssu-jira.softserveinc.com/browse/PROD-72

# Description

Created sign_up, sign_in using Devise gem
Added Bootstrap style
Created skip validation in User model
screenshot from 2018-04-10 12-56-01

# Checklist:

- [x] Rubocop doesn't raise any warnings and offenses
- [x] PR is approved by 2 team mates
- [x] PR is approved by 2 experts

### CODE REVIEW

File: app/views/devise/registrations/new.html.haml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.container
+  .row.justify-content-center
+    .col-lg-4.col-md-4
```
Comment (sashamuliar) [inline app/views/devise/registrations/new.html.haml]: Do you really need 'col-lg' class if you set 'col-md' class and they have equal number of columns?

**Review by sashamuliar — APPROVED** [review]

**Review by gnaizer — APPROVED** [review]

File: app/models/user.rb [hunk 2]
```diff
@@ -3,8 +3,12 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :trackable, :validatable
   validates :name, :surname, presence: true,
                              length: { in: 3..30 },
-                             format: { with: /\A[a-zA-Z]+\z/ }
+                             format: { with: /\A[a-zA-Z]+\z/ },
+                             unless: :skip_register_validation
```
Comment (okohuch) [inline app/models/user.rb]: do you need to skip all of the validations?

File: app/models/user.rb [hunk 3]
```diff
@@ -3,8 +3,12 @@ class User < ApplicationRecord
...
   validates :phone_number, presence: true,
                            numericality: true,
-                           length: { is: 13 }
+                           length: { is: 13 },
+                           unless: :skip_register_validation
```
Comment (okohuch) [inline app/models/user.rb]: same here

File: app/models/user.rb (hunk 2 above)
Comment (Pavliv) [reply app/models/user.rb]: screenshot from 2018-04-10 16-32-45
Without skipping I had errors for all fields.

File: app/models/user.rb (hunk 3 above)
Comment (Pavliv) [reply app/models/user.rb]: the same here
screenshot from 2018-04-10 16-32-45

File: app/controllers/users/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
```
Comment (viktoriia-bilous) [inline app/controllers/users/registrations_controller.rb]: Do you need this?

File: app/assets/stylesheets/application.scss [hunk 5]
```diff
@@ -0,0 +1 @@
+@import "bootstrap";
```
Comment (viktoriia-bilous) [inline app/assets/stylesheets/application.scss]: I thought bootstrap will be added in another PR

File: app/models/user.rb (hunk 2 above)
Comment (viktoriia-bilous) [reply app/models/user.rb]: If these fields are required then user should fill in all required fields during sign up

File: app/views/devise/registrations/new.html.haml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+              = t("#{@minimum_password_length} characters minimum")
+          %br
+          = f.password_field :password, autocomplete: "off", class: "form-control"
+        .form-group
+          = f.label :password_confirmation
+          %br
+          = f.password_field :password_confirmation, autocomplete: "off", class: "form-control"
+        .actions
+          = f.submit t(".sign_up"), class: "btn btn-outline-info"
+      = render "devise/shared/links"
```
Comment (viktoriia-bilous) [inline app/views/devise/registrations/new.html.haml:34]: Do you have this partial already merged?

**Comment by okohuch** [comment]
@Pavliv please address Vika's comment.

File: app/views/devise/registrations/new.html.haml (hunk 6 above)
Comment (Pavliv) [reply app/views/devise/registrations/new.html.haml:34]: No, this one Nastia did. So I deleted file (partial) and added only this line

Comment (viktoriia-bilous) [reply app/views/devise/registrations/new.html.haml:34]: If you merge before Nastia, it will crash
You can't add this

File: app/views/devise/sessions/new.html.haml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+          = f.label :password
+          %br
+          = f.password_field :password, autocomplete: "off", class: "form-control"
+        - if devise_mapping.rememberable?
+          .form-group
+            = f.check_box :remember_me
+            = f.label :remember_me
+        .actions
+          = f.submit t(".sign_in"), class: "btn btn-outline-info"
+      = render "devise/shared/links"
```
Comment (viktoriia-bilous) [inline app/views/devise/sessions/new.html.haml:20]: Same here

File: app/views/devise/registrations/new.html.haml (hunk 6 above)
Comment (omaik) [reply app/views/devise/registrations/new.html.haml:34]: rebase from Nastia?

Comment (Pavliv) [reply app/views/devise/registrations/new.html.haml:34]: without partial file it use "gem devise", after merge it will use Nastia`s partial, is it?
In my view it shouldn't crash.

**Review by viktoriia-bilous — APPROVED** [review]

File: app/controllers/users/registrations_controller.rb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+module Users
```
Comment (okohuch) [inline app/controllers/users/registrations_controller.rb:1]: why do you need to wrap it in a module?

Comment (Pavliv) [reply app/controllers/users/registrations_controller.rb:1]: when "devise" generate his controller it was
class Users::RegistrationsController < Devise::RegistrationsController
but rubocop didn't allow me use shortcut method, he asked about nested style.

**Review by okohuch — APPROVED** [review]
